### ISSUE TITLE
False positive related to local variable type inference from conditional statement

### ISSUE THREAD

**Issue by MichalStehlikCz** [typetools/checker-framework#6290]
After upgrade from 3.32.0 > 3.40.0, I get warnings on following code
```
  public Optional<String> test(@Nullable String param) {
    var first = Optional.ofNullable(param);
    var second = first.isPresent() ? first : Optional.ofNullable(param);
    return second;
  }
```
It now produces
```
java: [return] incompatible types in return.
  type of expression: @Initialized @NonNull Optional<@Initialized @Nullable String>
  method return type: @Initialized @NonNull Optional<@Initialized @NonNull String>
```
which I believe is false positive.

Equivalent code with both variables using explicit type
```
  public Optional<String> test(@Nullable String param) {
    Optional<String> first = Optional.ofNullable(param);
    Optional<String> second = first.isPresent() ? first : Optional.ofNullable(param);
    return second;
  }
```
produces no warnings
